petastorm/arrow_reader_worker.py

```diff
@@ -168,6 +170,38 @@ def process(self, piece_index, worker_predicate, shuffle_row_drop_partition):
         if all_cols:
             self.publish_func(all_cols)
 
+    def infer_schema_from_first_row(self):
```

Comment:
nit: I'm not sure whether the partition[0] necessarily contains the "first" row? Could the partitions be out of order? If so, we may call it `infer_schema_from_a_row`.

Reply:
Here I read the first row in the index-0 row-groups. But index-0 row-groups may be non-deterministic ? Not sure. `infer_schema_from_a_row` sounds good.